.ci/targets/ios_platform_tests.yaml

```diff
@@ -21,7 +21,11 @@ tasks:
     args: ["xcode-analyze", "--ios", "--ios-min-version=13.0"]
   - name: native test
     script: script/tool_runner.sh
-    args: ["native-test", "--ios", "--ios-destination", "platform=iOS Simulator,name=iPhone 14,OS=latest"]
+    args: ["native-test", "--ios", "--ios-destination", "platform=iOS Simulator,name=Flutter-iPhone,OS=16.4"]
```

Comment:
As we upgrade, there's no guarantee that there will be an iPhone 14 simulator. Better to use the simulator we create in the "create simulator" step (same for the pigeon test_suites.dart).

Reply:
Pigeon's `test_suites` is a weird case; we need it to run locally as well as in CI. So we don't want to hard-code it to a simulator that only exists in CI.

Reply:
Okay, so I updated Pigeon's `test_suites` to create and delete it's own simulator so that it'll work in CI and locally

FYI couple reasons I think that's the best approach: 
1.  Due to new way Simulator runtimes are handled, when you use `OS=latest` or you omit the OS (which defaults to `latest`), it may use a newer simulator runtime that's not the default for the Xcode version. For example, if iOS 17 runtime is mounted, it'll use that instead of iOS 16.4 even if you're using Xcode 14.3. So even if there's an "iPhone 14" device with OS 16.4, it won't match against it since it's looking for iOS 17.
2. There's no guarantee to be a Simulator named "iPhone 14" unless we create it beforehand (both in CI and locally)

Reply:
Also btw, I ran `dart run tool/run_tests.dart` in `packages/packages/pigeon` locally to make sure it passed locally